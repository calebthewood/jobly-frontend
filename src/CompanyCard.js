import { Link } from "react-router-dom";

/** Creates card for each company with a link
 *
 * prop - {company}
 *
 * state - none
 *
 * CompanyList -> CompanyCard
 */

function CompanyCard({ company }) {

  return (
    <Link to={`/companies/${company.handle}`}>
      <div className="row justify-content-center">

        <div className="card shadow mb-4">
          <div className="card-body">
            <h6 className="card-title">
              {company.name}
              {company.logoUrl &&
                <img
                  src={`${company.logoUrl}`}
                  alt={company.handle}
                  className="float-end"
                  style={{ height: "2rem" }} />}
            </h6>
            <p className="card-text fw-light">{company.description}</p>
          </div>
        </div>

      </div>
    </Link>
  );
}

export default CompanyCard;