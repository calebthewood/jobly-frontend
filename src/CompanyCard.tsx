import { Link } from "react-router-dom";
import { ICompanyCardProps } from "./interfaces";

/** Creates card for each company with a link
 *
 * prop - {company}
 *
 * state - none
 *
 * CompanyList -> CompanyCard
 */

function CompanyCard({ company }: ICompanyCardProps) {

  return (
    <Link to={`/companies/${company.handle}`}>
      <div className="row justify-content-center">
        <div className="col-12 col-md-10 col-xl-7">
          <div className="card shadow mb-4">
            <div className="card-body">
              <h6 className="card-title">
                {company.name}
                {company.logoUrl &&
                  <img
                    src={`${company.logoUrl}`}
                    alt={company.handle}
                    className="float-end"
                    style={{ height: "2rem" }} />}
              </h6>
              <p className="card-text fw-light">{company.description}</p>
            </div>
          </div>
        </div>
      </div>
    </Link>
  );
}

export default CompanyCard;