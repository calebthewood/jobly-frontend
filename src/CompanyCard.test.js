import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

const companyOne = {
  handle: "rithm",
  name: "Rithm School",
  description: "Become an exceptional developer in 16 weeks.",
  logo_url: "https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
};

const companyTwo = {
  handle: "algo",
  name: "Algo School",
  description: "Become a mediocre developer in 160 weeks."
}

it("matches snapshot with logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={companyOne} />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={companyTwo}/>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});