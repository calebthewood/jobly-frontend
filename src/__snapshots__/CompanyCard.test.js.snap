// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    href="/companies/rithm"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-12 col-md-10 col-xl-7"
      >
        <div
          class="card shadow mb-4"
        >
          <div
            class="card-body"
          >
            <h6
              class="card-title"
            >
              Rithm School
            </h6>
            <p
              class="card-text fw-light"
            >
              Become an exceptional developer in 16 weeks.
            </p>
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    href="/companies/algo"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-12 col-md-10 col-xl-7"
      >
        <div
          class="card shadow mb-4"
        >
          <div
            class="card-body"
          >
            <h6
              class="card-title"
            >
              Algo School
            </h6>
            <p
              class="card-text fw-light"
            >
              Become a mediocre developer in 160 weeks.
            </p>
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
